src/main/java/gregtech/common/metatileentities/storage/MetaTileEntityCreativeChest.java

```diff
@@ -77,45 +87,24 @@ public MetaTileEntity createMetaTileEntity(IGregTechTileEntity tileEntity) {
     }
 
     @Override
-    protected ModularUI createUI(EntityPlayer entityPlayer) {
-        ModularUI.Builder builder = ModularUI.builder(GuiTextures.BACKGROUND, 176, 209)
-                .bindPlayerInventory(entityPlayer.inventory, 126);
-        builder.widget(new PhantomSlotWidget(handler, 0, 36, 6)
-                .setClearSlotOnRightClick(true)
-                .setBackgroundTexture(GuiTextures.SLOT)
-                .setChangeListener(this::markDirty));
-        builder.label(7, 9, "gregtech.creative.chest.item");
-        builder.widget(new ImageWidget(7, 48, 154, 14, GuiTextures.DISPLAY));
-        builder.widget(new TextFieldWidget2(9, 50, 152, 10, () -> String.valueOf(itemsPerCycle), value -> {
-            if (!value.isEmpty()) {
-                itemsPerCycle = Integer.parseInt(value);
-            }
-        }).setMaxLength(11).setNumbersOnly(1, Integer.MAX_VALUE));
-        builder.label(7, 28, "gregtech.creative.chest.ipc");
-
-        builder.widget(new ImageWidget(7, 85, 154, 14, GuiTextures.DISPLAY));
-        builder.widget(new TextFieldWidget2(9, 87, 152, 10, () -> String.valueOf(ticksPerCycle), value -> {
-            if (!value.isEmpty()) {
-                ticksPerCycle = Integer.parseInt(value);
-            }
-        }).setMaxLength(11).setNumbersOnly(1, Integer.MAX_VALUE));
-        builder.label(7, 65, "gregtech.creative.chest.tpc");
-
-        builder.widget(new CycleButtonWidget(7, 101, 162, 20, () -> active, value -> {
-            active = value;
-            scheduleRenderUpdate();
-            var c = getQuantumController();
-            if (c != null) c.onHandlerUpdate();
-        }, "gregtech.creative.activity.off", "gregtech.creative.activity.on"));
-
-        builder.widget(createConnectedGui(6));
-
-        return builder.build(getHolder(), entityPlayer);
+    public ModularPanel buildUI(PosGuiData guiData, PanelSyncManager guiSyncManager) {
+        return appendCreativeUI(GTGuis.createPanel(this, 176, 166), false,
+                new BoolValue.Dynamic(() -> active, b -> active = b),
```

Comment:
The color of the text for Active/Not active blends with the button for the not active case.